drivers/usb/udc/Kconfig.mcux

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+config USB_UDC_NXP_EHCI
+	bool "NXP MCUX USB EHCI Device controller driver"
+	default y
+	depends on DT_HAS_NXP_EHCI_ENABLED
+	select USB_UDC_NXP_MCUX
+	help
+	  NXP MCUX USB Device Controller Driver for EHCI.
+
+config USB_UDC_NXP_IP3511
+	bool "NXP MCUX USB IP3511 Device controller driver"
+	default y
+	depends on DT_HAS_NXP_LPCIP3511_ENABLED
+	select USB_UDC_NXP_MCUX
+	help
+	  NXP MCUX USB Device Controller Driver for KHCI.
```

Comment:
I cannot find UDC driver implementations for controller EHCI and IP3511 in this PR. I think there is a misunderstanding. Is this a shim driver for a vendor USB middleware? Considering that Zephyr is there to fix all the issues with fancy vendor SDKs and provide a maintainable "infrastructure", this does not really look like a step forward. Can we please at least have a driver per controller type, that provides maintainable code for specific controller type and does not hide details in some other repository?

Reply:
Hi @jfischer-no Yes, it is a shim driver that uses the NXP SDK USB controller drivers. The SDK USB controller drivers are very stable because they are tested with many NXP SDK USB stack examples that are released many times in NXP SDK, so I prefer to use them. I have did some dependency simplification of NXP SDK USB controller drivers (https://github.com/zephyrproject-rtos/hal_nxp/pull/353) to use it more easily in Zephyr.
Because the SDK USB controller drivers provide the same interface, so I only create one udc_mcux.c for EHCI and IP3511. I can create two (udc_mcux_ehci,c and udc_mcux_ip3511.c) for them, but they will be almost same. I am OK if two is more  consistent with Zephyr style. Please let me know your thought.